grpc/server/test/connection_initiator_test.cpp

```diff
@@ -0,0 +1,84 @@
+#include <future>
+#include <gtest/gtest.h>
+
+#include "connection_initiator.h"
+#include "mocks/dronecore_mock.h"
+
+using testing::_;
+using testing::NiceMock;
+
+using event_callback_t = dronecore::testing::event_callback_t;
+using MockDroneCore = NiceMock<dronecore::testing::MockDroneCore>;
+using ConnectionInitiator = dronecore::backend::ConnectionInitiator<MockDroneCore>;
+
+static constexpr auto ARBITRARY_PORT = 1291;
+static constexpr auto ARBITRARY_UUID = 1492;
+
+ACTION_P(SaveCallback, event_callback)
+{
+    *event_callback = arg0;
+}
+
+TEST(ConnectionInitiator, registerDiscoverIsCalledExactlyOnce)
+{
+    ConnectionInitiator initiator;
+    MockDroneCore dc;
+    EXPECT_CALL(dc, register_on_discover(_))
+    .Times(1);
+
+    initiator.start(dc, ARBITRARY_PORT);
+}
+
+TEST(ConnectionInitiator, connectAddsUDPConnection)
+{
+    ConnectionInitiator initiator;
+    MockDroneCore dc;
+
+    EXPECT_CALL(dc, add_udp_connection(_));
+
+    initiator.start(dc, ARBITRARY_PORT);
+}
+
+TEST(ConnectionInitiator, connectHangsUntilDeviceDiscovered)
+{
+    ConnectionInitiator initiator;
+    MockDroneCore dc;
+    event_callback_t discover_callback;
+    EXPECT_CALL(dc, register_on_discover(_))
+    .WillOnce(SaveCallback(&discover_callback));
```

Comment:
This should be indented, but I can't seem to find how to do this with astyle -_-.

Reply:
Yer that's weird about astyle. I guess at some point we should try to out the clang tooling to enforce style.